doc/CloudIntegration.xml

```diff
@@ -0,0 +1,338 @@
+        The service is described using the YAML-based file. Then, the YAML file is used as the basis
+        for the service requester (OCS) implementation/integration. Integration is simplified
+        through the use of code generating tools that generate platform specific code that can be
+        used by the developers to integrate the Web Service.</para>
+      <para>The Web Service provider and requester communicate using the OpenAPI message exchange
+        protocol. OpenAPI is a lightweight messaging protocol used to encode the information in a
+        Web Service request and in a response message before sending them over a network. OpenAPI
+        messages are independent of any operating system or protocol and may be transported using a
+        variety of Internet protocols. This ONVIF standard defines conformant transport protocols
+        for the SOAP messages for the described Web Services.</para>
+      <para>The Web Service overview section introduces into the general ONVIF service structure,
+        the command definition syntax in the specification, error handling principles and the
+        adopted Web Service security mechanisms.</para>
+      </section>
+      <section xml:id="_Ref208379139">
+        <title>Security</title>
+        <section>
+          <title>Authentication</title>
+          <para>The services defined in this standard, whenever consumed, shall be protected using
+          digest authentication according to [RFC 2617] over HTTP and HTTPS or JWT client
+          authorization based on [RFC 6750], only over HTTPS. Additional information about HTTP
+          digest  authenticationcan can be found the ONVIF Core specification, while the JWT client
+          authorization can be found in the ONVIF Security Service.</para>
+          <para>MCS and OCS may support the SHA-256 hashing algorithm in HTTP and HTTPS. An ONVIF
+          compliant service should follow [RFC 7616].</para>
+        </section>
+        <section>
+          <title>User levels and access classes for the service</title>
+          <para>Since this service is only meant to enable sharing devices between different cloud
+          services, there is no need for defining different access classes to authorized access to
+          the functions defined in this specification.</para>
+        </section>
+      </section>
+    <section xml:id="_Ref208379149">
+      <title>Format of the share token</title>
+      <para>To start the device sharing operation, the OCS must provide a share token to the PCS, to
+        prove it has the rights to access the camera. They way the OCS retrieves the share token is
+        beyond the scope of this specification.</para>
+      <para>The share token must be a JWT and must include the <emphasis>xaddr</emphasis> claim that
+        will include the URI where the PCS will be accepting incoming requests from the OCS.</para>
+      </section>
+  </chapter>
+  <chapter>
+    <title>Device sharing</title>
```

Comment:
If an end-user wants to start device sharing so they send a transfer token to the OCS, but then change its mind (There's an issue with this specific device, they had a mixup from their tokens and it's for another devices, etc). Should we have a way to cancel a pending transfer? Where the OCS can send the same token and request cancellation of the sharing?

The PCS would be free to allow the cancellation if it is early enough (the device hasn't reached up yet/is offline) or deny it if the transfer is in progress and will finish (then, the callback would be called, and the OCS can handle it from there).

Reply:
Could this be discussed after we release the first revision of the spec?

Reply:
I agree with @jmelancongen there is no defined work flow to remove a device after adding to an OCS. 
If a device is able to be removed from an OCS where does it register itself with the original MCS ? or lay in limbo? 

Are we advocating Factory Resetting of a Device to remove it from an OCS and point back at an MCS but that would leave the OCS looking for a device it should have and Polling for it repeatedly as being removed.

Further to this what if a device just goes offline or is replaced as it went faulty ?